File: Tests/TestData/MultiFileCharacterization/SourceFiles/WindowsAppVb/WinformsDesignerTest.vb
Public Class WinformsDesignerTest
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

    End Sub

    Private Sub CheckedChangedOrButtonClicked(sender As Object, e As EventArgs) Handles CheckBox1.CheckedChanged, Button1.Click
        Dim formConstructedText  = "Form constructed"
        If Not(My.Forms.WinformsDesignerTest Is Nothing) AndAlso My.Forms.WinformsDesignerTest.Text <> formConstructedText Then
            My.Forms.WinformsDesignerTest.Text = formConstructedText
        Else If My.Forms.WinformsDesignerTest IsNot Nothing AndAlso DirectCast(My.Forms.WinformsDesignerTest, WinformsDesignerTest) IsNot Nothing
            My.Forms.WinformsDesignerTest = Nothing
        End If
    End Sub

    Private Sub WinformsDesignerTest_EnsureSelfEventsWork(sender As Object, e As EventArgs) Handles MyBase.Load, Me.SizeChanged

    End Sub

    Private Sub WinformsDesignerTest_MouseClick() Handles Me.MouseClick

    End Sub

    Private Sub ButtonMouseClickWithNoArgs() Handles Button1.MouseClick, Button2.MouseClick

    End Sub

    Private Sub ButtonMouseClickWithNoArgs2() Handles Button1.MouseClick, CheckBox1.CheckedChanged

    End Sub

    Public Sub Init()
        Dim noArgs As MouseEventHandler = AddressOf WinformsDesignerTest_MouseClick
        AddHandler Me.MouseClick, noArgs
        AddHandler Me.MouseClick, AddressOf WinformsDesignerTest_MouseClick
        RemoveHandler Me.MouseClick, noArgs
        RemoveHandler Me.MouseClick, AddressOf WinformsDesignerTest_MouseClick ' Generates a VB warning because it has no effect
    End Sub

    Public Sub Init_Advanced(paramToHandle As MouseEventHandler)
        Init()
        AddHandler Me.MouseClick, paramToHandle
        WinformsDesignerTest_MouseClick()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        FolderForm.ShowDialog()
    End Sub
End Class

File: Tests/TestData/MultiFileCharacterization/VBToCSResults/ConvertWholeSolution/WindowsAppVb/WinformsDesignerTest.cs
using System;
using System.Windows.Forms;

namespace WindowsAppVb
{
    public partial class WinformsDesignerTest
    {
        public WinformsDesignerTest()
        {
            InitializeComponent();
            _Button1.Name = "Button1";
            _CheckBox1.Name = "CheckBox1";
            _Button2.Name = "Button2";
        }

        private void Button1_Click(object sender, EventArgs e)
        {
        }

        private void CheckedChangedOrButtonClicked(object sender, EventArgs e)
        {
            string formConstructedText = "Form constructed";
            if (My.MyProject.Forms.m_WinformsDesignerTest is object && (My.MyProject.Forms.WinformsDesignerTest.Text ?? "") != (formConstructedText ?? ""))
            {
                My.MyProject.Forms.WinformsDesignerTest.Text = formConstructedText;
            }
            else if (My.MyProject.Forms.m_WinformsDesignerTest is object && My.MyProject.Forms.m_WinformsDesignerTest is object)
            {
                My.MyProject.Forms.WinformsDesignerTest = null;
            }
        }

        private void WinformsDesignerTest_EnsureSelfEventsWork(object sender, EventArgs e)
        {
        }

        private void WinformsDesignerTest_MouseClick()
        {
        }

        private void ButtonMouseClickWithNoArgs()
        {
        }

        private void ButtonMouseClickWithNoArgs2()
        {
        }

        public void Init()
        {
            MouseEventHandler noArgs = (_, __) => WinformsDesignerTest_MouseClick();
            MouseClick += noArgs;
            MouseClick += (_, __) => WinformsDesignerTest_MouseClick();
            MouseClick -= noArgs;
            MouseClick -= (_, __) => WinformsDesignerTest_MouseClick(); // Generates a VB warning because it has no effect
        }

        public void Init_Advanced(MouseEventHandler paramToHandle)
        {
            Init();
            MouseClick += paramToHandle;
            WinformsDesignerTest_MouseClick();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.FolderForm.ShowDialog();
        }

        private void WinformsDesignerTest_MouseClick(object sender, MouseEventArgs e) => WinformsDesignerTest_MouseClick();
        private void ButtonMouseClickWithNoArgs(object sender, MouseEventArgs e) => ButtonMouseClickWithNoArgs();
        private void ButtonMouseClickWithNoArgs2(object sender, MouseEventArgs e) => ButtonMouseClickWithNoArgs2();
        private void ButtonMouseClickWithNoArgs2(object sender, EventArgs e) => ButtonMouseClickWithNoArgs2();
    }
}